packages/react/src/components/table/Table.stories.tsx

```diff
@@ -0,0 +1,934 @@
+      <DataTable checkboxSelection dense caption={caption} cellConfig={cellConfig} rows={rows} />
+    </div>
+  );
+};
+
+// args is required for docs tab to show source code
+// eslint-disable-next-line no-unused-vars
+export const InitiallySelectedRows = (args) => {
+  const cellConfig = {
+    cols: [
+      { key: 'id', headerName: 'Not rendered' },
+      { key: 'firstName', headerName: 'First name' },
+      { key: 'surname', headerName: 'Surname' },
+      {
+        key: 'age',
+        headerName: 'Age',
+        transform: ({ age }) => {
+          return <div style={{ textAlign: 'right' }}>{age}</div>;
+        },
+      },
+      { key: 'profession', headerName: 'Profession' },
+    ],
+    indexKey: 'id',
+    renderIndexCol: false,
+  };
+
+  const rows: Array<object> = [
+    { id: 1000, firstName: 'Lauri', surname: 'Kekkonen', age: 39, profession: 'Engineer' },
+    { id: 1001, firstName: 'Maria', surname: 'Sarasoja', age: 62, profession: 'Designer' },
+    { id: 1002, firstName: 'Anneli', surname: 'Routa', age: 50, profession: 'Meteorologist' },
+    { id: 1003, firstName: 'Osku', surname: 'Rausku', age: 18, profession: 'Mail Carrier' },
+  ];
+
+  const initiallySelectedRows = [1002, 1003];
+
+  const caption = (
+    <span>
+      <b>Table 1</b>: Table description
+    </span>
+  );
+
+  return (
+    <div style={{ maxWidth: '640px' }}>
+      <DataTable
+        caption={caption}
+        initiallySelectedRows={initiallySelectedRows}
+        checkboxSelection
+        variant="dark"
+        cellConfig={cellConfig}
+        rows={rows}
+      />
+    </div>
+  );
+};
+
+// args is required for docs tab to show source code
+// eslint-disable-next-line no-unused-vars
+export const SelectAllRows = (args) => {
+  const cellConfig = {
+    cols: [
+      { key: 'id', headerName: 'Not rendered' },
+      { key: 'firstName', headerName: 'First name' },
+      { key: 'surname', headerName: 'Surname' },
+      {
+        key: 'age',
+        headerName: 'Age',
+        transform: ({ age }) => {
+          return <div style={{ textAlign: 'right' }}>{age}</div>;
+        },
+      },
+      { key: 'profession', headerName: 'Profession' },
+    ],
+    indexKey: 'id',
+    renderIndexCol: false,
+  };
+
+  const rows: Array<object> = [
+    { id: 1000, firstName: 'Lauri', surname: 'Kekkonen', age: 39, profession: 'Engineer' },
+    { id: 1001, firstName: 'Maria', surname: 'Sarasoja', age: 62, profession: 'Designer' },
+    { id: 1002, firstName: 'Anneli', surname: 'Routa', age: 50, profession: 'Meteorologist' },
+    { id: 1003, firstName: 'Osku', surname: 'Rausku', age: 18, profession: 'Mail Carrier' },
+  ];
+
+  const [tableRows, setTableRows] = useState(rows);
+  const [selectedRows, setSelectedRows] = useState([]);
+
+  const myRef = useRef();
+
+  return (
+    <div style={{ maxWidth: '640px' }}>
+      <div style={{ marginBottom: '16px', display: 'flex', flexDirection: 'row', justifyContent: 'flex-end' }}>
+        <Button
```

Comment:
Should these 'Select all rows', 'Deselect all rows' and 'Delete selected' buttons be a part of DataTable implementation, or should we leave them out as they currently are?

Reply:
Could the selected be part of the row-data? 🤔 
Then it could be completely outside of the Table component. And initallySelectedRows would be unnecessary

Reply:
We discussed this and it turned out to be hard/problematic.